pkg/service/controller_test.go

```diff
@@ -6,14 +6,324 @@ import (
 
 	"github.com/container-storage-interface/spec/lib/go/csi"
 	"github.com/ovirt/csi-driver/pkg/service"
+	ovirtclient "github.com/ovirt/go-ovirt-client"
 )
 
 func TestVolumeCreation(t *testing.T) {
 	helper := getMockHelper(t)
 	controller := service.NewOvirtCSIDriver(helper.GetClient(), "test")
+
+	createVolumeResponse, err := createTestVolume(helper, controller)
+	if err != nil {
+		t.Fatalf("failed to create test volume (%v)", err)
+	}
+	if createVolumeResponse.Volume.CapacityBytes < 4096 {
+		t.Fatalf("created volume is too small (%v)", err)
+	}
+	diskList, err := helper.GetClient().ListDisks()
+	if err != nil {
+		t.Fatalf("failed to list disks (%v)", err)
+	}
+	if len(diskList) != 1 {
+		t.Fatalf("incorrect number of disks created (%d)", len(diskList))
+	}
+	disk := diskList[0]
+	if disk.TotalSize() < 4096 {
+		t.Fatalf("incorrect disk size on the backend: %d", disk.TotalSize())
+	}
+}
+
+func TestDeleteVolume(t *testing.T) {
+	helper := getMockHelper(t)
+	controller := service.NewOvirtCSIDriver(helper.GetClient(), "test")
+	createVolumeResponse, err := createTestVolume(helper, controller)
+	if err != nil {
+		t.Fatalf("failed to create test volume (%v)", err)
+	}
+
+	_, err = controller.DeleteVolume(context.Background(), &csi.DeleteVolumeRequest{
+		VolumeId: createVolumeResponse.Volume.VolumeId,
+	})
+	if err != nil {
+		t.Fatalf("failed to create deleteVolumeResponse (%v)", err)
```

Comment:
Sorry to poke, this is also in need of a better error message. Our future self will thank us if this test fails.

Reply:
np, my fault. Should be fixed now.